evalml/tests/component_tests/test_exponential_smoothing_regressor.py

```diff
@@ -0,0 +1,202 @@
+from unittest.mock import patch
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from evalml.model_family import ModelFamily
+from evalml.pipelines.components import ExponentialSmoothingRegressor
+from evalml.problem_types import ProblemTypes
+
+pytestmark = [
+    pytest.mark.noncore_dependency,
+    pytest.mark.skip_during_conda,
+    pytest.mark.skip_if_39,
+]
+
+
+def test_model_family():
+    assert (
+        ExponentialSmoothingRegressor().model_family
+        == ModelFamily.EXPONENTIAL_SMOOTHING
+    )
+
+
+def test_problem_types():
+    assert set(ExponentialSmoothingRegressor.supported_problem_types) == {
+        ProblemTypes.TIME_SERIES_REGRESSION
+    }
+
+
+def test_model_instance(ts_data):
+    X, y = ts_data
+    clf = ExponentialSmoothingRegressor()
+    fitted = clf.fit(X, y)
+    assert isinstance(fitted, ExponentialSmoothingRegressor)
+
+
+def test_fit_ts_without_y(ts_data):
+    X, y = ts_data
+
+    clf = ExponentialSmoothingRegressor()
+    with pytest.raises(
+        ValueError, match="Exponential Smoothing Regressor requires y as input."
+    ):
+        clf.fit(X=X)
+
+
+@pytest.mark.parametrize("train_features_index_dt", [True, False])
+@pytest.mark.parametrize("train_target_index_dt", [True, False])
+@pytest.mark.parametrize(
+    "train_none, no_features, datetime_feature",
+    [
+        (True, False, False),
+        (False, True, False),
+        (False, False, True),
+        (False, False, False),
+    ],
+)
+def test_remove_datetime(
```

Comment:
Minor nit, but it might be nice to add a patch on the component's fit to ensure that it is being passed the y data we expect (`call_args`). Not necessary, but would be nice to ensure this.

Reply:
Agreed. If the intention is to test that `fit` removes the datetime, it'd be better to mock the _component_obj.fit rather than testing the internal method directly?